GitUpKit/Core/GCDiff.m

```diff
@@ -354,6 +354,27 @@ - (void)_cacheDeltasIfNeeded {
         XLOG_DEBUG_UNREACHABLE();
       }
     }
+
+    // Remove superfluous "untracked" deltas for conflicting submodules
+    // Needed when the input _deltas looks like this:
+    // 1: [Conflicted] "submodule"
+    // 2: [Untracked] "submodule/"
+    // Which happens every time there's a submodule entry that's a conflict
+    NSMutableArray<GCDiffDelta *> *deltasToFilterOut = [NSMutableArray array];
```

Comment:
I haven't used submodules in a number of years so I'll play around with this a bit just to have a clear understanding myself. I'll merge it in regardless for the time being. Would be nice to cut a new release with the fixes you have here so far.